app/models/api_token.rb

```diff
@@ -1,17 +1,26 @@
 class ApiToken < ActiveRecord::Base
   before_create :generate_token
-  validates_presence_of :user_id, :cart_id
-  belongs_to :cart
-  belongs_to :user
+
+  belongs_to :approval
+  has_one :cart, through: :approval
+  has_one :user, through: :approval
+
+  validates_presence_of :approval_id
 
   scope :unexpired, -> { where('expires_at >= ?', Time.now) }
   scope :expired, -> { where('expires_at < ?', Time.now) }
   scope :unused, -> { where(used_at: nil) }
   scope :fresh, -> { unused.unexpired }
 
+  def cart_id
+    self.approval.cart_id
+  end
+
   def generate_token
```

Comment:
I just noticed we don't have any specs for ApiToken. This method looks spec worthy.